File: implementation/scipamato/scipamato-core-web/src/test/java/ch/difty/scipamato/core/web/paper/jasper/PaperDataSourceTest.java
package ch.difty.scipamato.core.web.paper.jasper;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.verifyNoMoreInteractions;

import net.sf.jasperreports.engine.JRDataSource;
import net.sf.jasperreports.engine.JRException;
import net.sf.jasperreports.engine.design.JRDesignField;
import org.junit.After;
import org.mockito.Mock;

import ch.difty.scipamato.core.entity.Paper;
import ch.difty.scipamato.core.entity.search.PaperFilter;
import ch.difty.scipamato.core.entity.search.SearchOrder;
import ch.difty.scipamato.core.web.WicketTest;
import ch.difty.scipamato.core.web.paper.AbstractPaperSlimProvider;

public abstract class PaperDataSourceTest extends WicketTest {

    @Mock
    protected AbstractPaperSlimProvider<PaperFilter> dataProviderMock;
    @Mock
    protected PaperFilter                            paperFilterMock;
    @Mock
    protected SearchOrder                            searchOrderMock;
    @Mock
    protected Paper                                  paperMock;
    @Mock
    protected ClusterablePdfExporterConfiguration    pdfExporterConfigMock;
    @Mock
    protected CoreShortFieldConcatenator             shortFieldConcatenatorMock;

    protected PaperFilter paperFilter = new PaperFilter();

    @After
    public final void tearDown() {
        verifyNoMoreInteractions(dataProviderMock, paperFilterMock, searchOrderMock, paperMock,
            shortFieldConcatenatorMock);
    }

    protected void assertFieldValue(String fieldName, String value, JRDesignField f, final JRDataSource jsds)
        throws JRException {
        f.setName(fieldName);
        assertThat(jsds.getFieldValue(f)).isEqualTo(value);
    }

}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/paper/jasper/PaperDataSourceTest.kt
package ch.difty.scipamato.core.web.paper.jasper

import ch.difty.scipamato.core.entity.Paper
import ch.difty.scipamato.core.entity.search.PaperFilter
import ch.difty.scipamato.core.entity.search.SearchOrder
import ch.difty.scipamato.core.web.WicketTest
import ch.difty.scipamato.core.web.paper.AbstractPaperSlimProvider
import io.mockk.confirmVerified
import io.mockk.impl.annotations.MockK
import net.sf.jasperreports.engine.JRDataSource
import net.sf.jasperreports.engine.JRException
import net.sf.jasperreports.engine.design.JRDesignField
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.AfterEach

abstract class PaperDataSourceTest : WicketTest() {

    @MockK
    protected lateinit var dataProviderMock: AbstractPaperSlimProvider<PaperFilter>

    @MockK
    protected lateinit var paperFilterMock: PaperFilter

    @MockK
    protected lateinit var searchOrderMock: SearchOrder

    @MockK(relaxed = true)
    protected lateinit var paperMock: Paper

    @MockK
    protected lateinit var pdfExporterConfigMock: ClusterablePdfExporterConfiguration

    @MockK
    protected lateinit var shortFieldConcatenatorMock: CoreShortFieldConcatenator

    @AfterEach
    fun tearDown() {
        confirmVerified(dataProviderMock, paperFilterMock, searchOrderMock, paperMock, shortFieldConcatenatorMock)
    }

    @Throws(JRException::class)
    protected fun assertFieldValue(fieldName: String?, value: String?, f: JRDesignField, jsds: JRDataSource) {
        f.name = fieldName
        jsds.getFieldValue(f) shouldBeEqualTo value
    }
}
